test/parallel/test-fs-open.js

```diff
@@ -32,8 +32,20 @@ fs.open(__filename, 'rs', function(err, fd) {
   openFd2 = fd;
 });
 
+try {
+  fs.open('/path/to/file/that/does/not/exist', { bad: 'flags' }, function() {
+    throw new Error('fs.open callback should not have been called');
+  });
+
+  throw new Error('fs.open did not throw despite bad flags');
+} catch (err) {
+  if (!(err instanceof TypeError)) {
+    throw new Error('unexpected error: ' + err);
+  }
+}
```

Comment:
I wanted to make sure it's a TypeError and not the Error of "bad flag value" (which can also happen when it's a string, just not the right one).